File: frontend/src/types/validation.ts
export type ValidationIssue = {
  code: string;
  message: string;
  category: string;
  sheet?: string | null;
  cell?: string | null;
  expected?: unknown;
  actual?: unknown;
  severity?: "ERROR" | "WARNING" | "REVIEW";
};

export type ReviewClassification = "likely_valid" | "likely_issue" | "formatting_variation" | "unverifiable" | "needs_human_review";

export type AIReviewClassification = {
  classification?: ReviewClassification;
  confidence?: number;
  reasoning?: string;
  suggested_action?: string;
  model_used?: string;
  latency_ms?: number;
  error?: string;
};

export type AIReviewSummary = {
  enabled: boolean;
  classifications: AIReviewClassification[];
  total_review_items: number;
  successfully_classified: number;
  failed_classifications: number;
};

export type FileValidationResult = {
  filename: string;
  status: "PASS" | "ERROR" | "REVIEW";
  template?: string | null;
  errors: ValidationIssue[];
  warnings: ValidationIssue[];
  review_items?: ValidationIssue[];
  ai_review?: AIReviewSummary;
};

export type ValidationSummary = {
  correct: number;
  incorrect: number;
  review: number;
  percentage_correct: number;
};

export type ValidationResponse = {
  total_files: number;

  summary: ValidationSummary;

  correct_files: string[];

  incorrect_files: FileValidationResult[];

  review_files: FileValidationResult[];

  files: FileValidationResult[];
};

export type HistoryRecord = {
  filename: string;
  timestamp: string;
  status: "PASS" | "ERROR" | "REVIEW";
  template?: string | null;
  error_count: number;
  review_count: number;
  validation_result: FileValidationResult;
  review_state: "PENDING" | "VERIFIED" | "INCORRECT";
  ai_review?: AIReviewSummary;
};

export type HistoryResponse = {
  history: HistoryRecord[];
  total: number;
};

export type ReviewStateFile = {
  filename: string;
  timestamp: string;
  status: "PASS" | "ERROR" | "REVIEW";
  template?: string | null;
  error_count: number;
  review_count: number;
  review_state: "PENDING" | "VERIFIED" | "INCORRECT";
  validation_result: FileValidationResult;
  ai_review?: AIReviewSummary;
};

export type ReviewStateResponse = {
  pending_files: ReviewStateFile[];
  total: number;
};

File: backend/app/models/validation.py
from pydantic import BaseModel
from typing import Any, Optional
from enum import Enum


class ValidationIssue(BaseModel):
    code: str
    message: str
    category: str
    sheet: Optional[str] = None
    cell: Optional[str] = None
    expected: Any = None
    actual: Any = None


class ReviewClassification(str, Enum):
    """Classification types for REVIEW items from AI classifier."""
    LIKELY_VALID = "likely_valid"
    LIKELY_ISSUE = "likely_issue"
    FORMATTING_VARIATION = "formatting_variation"
    UNVERIFIABLE = "unverifiable"
    NEEDS_HUMAN_REVIEW = "needs_human_review"


class AIReviewClassification(BaseModel):
    """AI classification result for a review item."""
    classification: Optional[ReviewClassification] = None
    confidence: Optional[float] = None
    reasoning: Optional[str] = None
    suggested_action: Optional[str] = None
    model_used: Optional[str] = None
    latency_ms: Optional[int] = None
    error: Optional[str] = None


class AIReviewSummary(BaseModel):
    """Summary of AI classification for a workbook."""
    enabled: bool
    classifications: list[AIReviewClassification] = []
    total_review_items: int = 0
    successfully_classified: int = 0
    failed_classifications: int = 0


class FileValidationResult(BaseModel):
    filename: str
    status: str
    errors: list[ValidationIssue] = []
    warnings: list[ValidationIssue] = []
    review_items: list[ValidationIssue] = []
    ai_review: Optional[AIReviewSummary] = None
